build/platform_and_feature_flags.props

```diff
@@ -2,7 +2,7 @@
   <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNetCore)' or '$(TargetFramework)' == '$(TargetFrameworkNet6)'">
     <DefineConstants>$(DefineConstants);NET_CORE;SUPPORTS_CONFIDENTIAL_CLIENT;SUPPORTS_CUSTOM_CACHE;SUPPORTS_BROKER;SUPPORTS_WIN32;</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNet6)' or '$(TargetFramework)' == '$(TargetFrameworkNet6Win)'">
+  <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNet6)' or '$(TargetFramework)' == '$(TargetFrameworkNet6Win)' or '$(TargetFramework)' == '$(TargetFrameworkNet7)' or '$(TargetFramework)' == '$(TargetFrameworkNet7Win)'">
```

Comment:
I don't think this is needed. MSAL doesn't explicitly target net7 and our tests don't need to know what JSON parsing lib MSAL is using?

Reply:
This is primarily for the unit tests to use SUPPORTS_SYSTEM_TEXT_JSON as a directive for .net 7. Technically it should be in the unit test project, Ill move it there